### DISCUSSION TITLE
What is the maximum JDK version does orientdb 2.2.34 community version supports?

### DISCUSSION THREAD

**Discussion by arpitdd123** [orientechnologies/orientdb#10017] [Q&A] [answered]
What is the maximum JDK version does orientdb 2.2.34 community version supports?

**Comment by tglman** [comment, marked as answer]
Hi,
The 2.2.x series is long deprecated please use a 3.2.x version.
to answer your question, I think back in time we did support 1.6+, with 3.2.x we still guarantee 1.8+

**Comment by andrii0lomakin** [comment]
Hi, 3.2.x currently does not support JDK 21, but there is WIP to provide support for JDK 21 soon.

**Reply by tglman** [reply]
Actually working on it, I noticed that to make work some dependencies on java 21, we need to drop the support of java 8 (because newer version of that lib do not work with java 8, and the one that work with java 8 do not work with java 21), so probably 3.2.x may not support in short term java 21 a newer version of OrientDB may drop java 8 and support java 21.

**Reply by OndraZizka** [reply]
Hi, I am all for dropping the ancient Java 1.8 in favor of at least 21. Using it for JBake.

What is the status of this? Thanks

**Reply by tglman** [reply]
so stable version (3.2.x) is tested till 17, the development version is already tested up to java 22 (actually from 17 to 22), and soon I may add testing for newer jdk version as well hopefully next year we will start to see the first new version out
